packages/models/src/actor/actor-reference.ts

```diff
@@ -51,11 +72,28 @@ export class ActorReference {
     this.#protocol = protocol
     this.#path = path
 
-    if (this.#path !== '/' && !/^\/\w+\/$/.test(this.#path)) {
+    if (this.#path !== '/' && !/^\/([\w:]+\/)*$/.test(this.#path)) {
       throw new InvalidPathException(this.#path)
     }
 
     this.#uri = this.#protocol + ':/' + this.#path + this.#name.toString()
+
+    this.#params = this.#path.split('/').reduce((params, param, index) => {
+      if (param.startsWith(':')) {
+        param = param.slice(1)
+        return params.set(param, { param, index })
+      }
+      return params
+    }, new Map())
+  }
+
+  matchParams(pattern: ActorRef): Map<string, ActorRefParam> {
+    const paths = this.#path.split('/')
+    pattern.params.forEach((param) => {
+      this.#params.set(param.param, { ...param, value: paths[param.index] })
+    })
+
+    return this.#params
```

Comment:
This function has been implemented in https://github.com/ckb-js/kuai/blob/5b6444870698254c251ce53db50ecc847bf16a3b/packages/io/src/router.ts#L116-L120 and can be extracted

Reply:
A concise one https://github.com/ckb-js/kuai/pull/191/files#diff-40b8e3199522c2c515d1865b62e7bbc4eb2c9e0654472a97baceb9a7306daec7R20-R22